wandering-warriors/main.py

```diff
@@ -27,9 +33,71 @@ class Ledger(Widget):
 class OperationsBar(Widget):
     pass
 
-
-class CuneiformDrawingInput(Widget):
-    pass
+def calculate_points(x1, y1, x2, y2, steps=1):
+    dx = x2 - x1
+    dy = y2 - y1
+    dist = sqrt(dx * dx + dy * dy)
+    if dist < steps:
+        return
+    o = []
+    m = dist / steps
+    for i in range(1, int(m)):
+        mi = i / m
+        lastx = x1 + dx * mi
+        lasty = y1 + dy * mi
+        o.extend([lastx, lasty])
+    return o
+
+class CuneiformDrawingInput(FloatLayout):
+
+    def on_touch_down(self, touch):
+        ud = touch.ud
+        if(touch.pos[0] > self.pos[0] and 
+           touch.pos[0] < self.pos[0] + self.size[0] and 
+           touch.pos[1] > self.pos[1] and 
+           touch.pos[1] < self.pos[1] + self.size[1]):
+                ud['group'] = g = str(touch.uid)
+                pointsize = 1
+                ud['color'] = 0
+                with self.canvas:
+                    Color(0, 0, 0)
+                    ud['lines'] = [
+                        Point(points=(touch.x, touch.y), source='particle.png',
+                              pointsize=pointsize, group=g)]
+                return True
+        else:
+            ud['lines'] = []
+
+    def on_touch_move(self, touch):
+        try:
+            if(touch.pos[0] > self.pos[0] and 
+               touch.pos[0] < self.pos[0] + self.size[0] and 
+               touch.pos[1] > self.pos[1] and 
+               touch.pos[1] < self.pos[1] + self.size[1]):
+                    ud = touch.ud
+
+                    points = ud['lines'][-1].points
+                    oldx, oldy = points[-2], points[-1]
+                    points = calculate_points(oldx, oldy, touch.x, touch.y)
+                    if points:
+                        try:
+                            lp = ud['lines'][-1].add_point
+                            for idx in range(0, len(points), 2):
+                                lp(points[idx], points[idx + 1])
+                        except GraphicException:
+                            pass
+        except:
```

Comment:
I'm not sure an empty `except` passes pep8... I don't know though.

Mind specifying which error we are `except`-ing for?

Reply:
That was just to get it to work. I know it's bad, I should.fix it tonight.